modules/make_first_decision.nf

```diff
@@ -0,0 +1,46 @@
+process MAKE_FIRST_DECISION {
+  label 'light'
+
+  input:
+  tuple val(id), val(length), val(dmcPrediction), val(tiaraPrediction)
+
+  output:
+  path "output.make_first_decision" 
+  
+  script:
+  // Cases where DeepMicroClass and Tiara have the same prediction
+    // Classified as eukaryote
+  if (dmcPrediction == "Eukaryote" && tiaraPrediction == "eukarya") 
+    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tClassifiedAsEukaryote' > output.make_first_decision"
+    // Classified as prokaryote
+  else if (dmcPrediction == "Prokaryote" && tiaraPrediction == "bacteria") 
+    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tClassifiedAsProkaryote' > output.make_first_decision"
+  else if (dmcPrediction == "Prokaryote" && tiaraPrediction == "prokarya")
+    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tClassifiedAsProkaryote' > output.make_first_decision"
+  else if (dmcPrediction == "Prokaryote" && tiaraPrediction == "archea") 
+    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tClassifiedAsProkaryote' > output.make_first_decision"
+  else if (dmcPrediction == "Plasmid" && tiaraPrediction == "bacteria") 
+    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tClassifiedAsProkaryote' > output.make_first_decision"
+  else if (dmcPrediction == "Plasmid" && tiaraPrediction == "prokarya")
+    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tClassifiedAsProkaryote' > output.make_first_decision"
+  else if (dmcPrediction == "Plasmid" && tiaraPrediction == "archea") 
+    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tClassifiedAsProkaryote' > output.make_first_decision"
+  
+  // Cases where Tiara classify as an eukaryote but DeepMicroClass don't
+  else if (dmcPrediction != "Eukaryote" && tiaraPrediction == "eukarya") 
+    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tClassifiedAsEukaryote' > output.make_first_decision"
+
+  // Cases where DeepMicroClass classify as a prokaryote but Tiara don't
+  else if (dmcPrediction == "Eukaryote" && tiaraPrediction == "archea") 
+    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tRequiringCATvalidation' > output.make_first_decision"
+  else if (dmcPrediction == "Eukaryote" && tiaraPrediction == "bacteria") 
+    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tRequiringCATvalidation' > output.make_first_decision"
+  else if (dmcPrediction == "Eukaryote" && tiaraPrediction == "prokarya")
+    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tRequiringCATvalidation' > output.make_first_decision"
```

Comment:
```suggestion
dmc_prok = ["Prokaryote", "Plasmid"]
tiara_prok = ["prokarya", "archaea", "bacteria"]

  // Cases where DeepMicroClass and Tiara have the same prediction
    // Classified as eukaryote
  if (dmcPrediction == "Eukaryote" && tiaraPrediction == "eukarya") 
    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tClassifiedAsEukaryote' > output.make_first_decision"
    
    // Classified as prokaryote
  else if (dmcPrediction in dmc_prok && tiaraPrediction in tiara_prok) 
    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tClassifiedAsProkaryote' > output.make_first_decision"
  
  // Cases where Tiara classify as an eukaryote but DeepMicroClass don't
  else if (dmcPrediction != "Eukaryote" && tiaraPrediction == "eukarya") 
    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tClassifiedAsEukaryote' > output.make_first_decision"

  // Cases where DeepMicroClass classify as a eukaryote but Tiara don't
  else if (dmcPrediction == "Eukaryote" && tiaraPrediction in tiara_prok) 
    "echo -e '$id\t$length\t$tiaraPrediction\t$dmcPrediction\tRequiringCATvalidation' > output.make_first_decision"
```

Couple questions - DMC euk/prok I assume also means viral for both?
Why is Tiara splitting prokarya from archaea and bacteria? Is there some confidence cutoff for the more specific predictions or does prokarya mean something else?

Reply:
I think the code could probably be simplified even further but please do check that I didn't mess up the logic. I haven't tested it.

Reply:
1. No it doesn't, that's why I was waiting for the viral classification test but according to the results I think it's better to ignore the virus scores and use the eukaryotes, prokaryotes and plasmids ones for this first step. 
2. The prokarya class is used when Tiara can't distinguish archea and bacteria. There is also the organella class, which can be ambiguous because of some organella origin. 
3. I indeed need to also add this case.

I'll write all the possibilities and add them to this script, always minimizing categorization as eukaryotic to avoid misclass of prokaryotic contigs. About the code itself, I don't know if it's possible as I never saw "in" used in Nextflow for the moment, but as I'm taking all these advices one by one I will can check it, thanks !